### ISSUE TITLE
How to query current active block on runtime, and what blocks it leads to?

### ISSUE THREAD

**Issue by TheEmbracedOne** [snozbot/fungus#863]
I would like to create a custom C# script (or BOLT state machine) that would need to “know” which block is currently the active one in a Fungus Flowchart on runtime, and what blocks it leads to. 

The following is what I'd like to achieve:
- [X] 1) Get the currently active Block
- [X] 2) Query the Menu commands in it to see where they "lead" and query their block descriptions
- [X] 3) Return a List of strings, paired as "BlockName" and "BlockDescriptions"
- [x] 4) Upon selecting an option, select a new active block based on block descriptions instead **<- this is what I'm having a problem with**

I would like to use this information to determine what menu option the player chose to click, at the very moment they click it, then based on the description of the new active block, select another block to be the new active block instead. It is important for what I'm trying to achieve here to have this kind of "foresight". Some more context:

I have a custom stat manager system that I use to modify player stats on almost every single block via Invoke Event.
Each block has a description that acts as a tag, to note what stat is being modified in that block.
I would like to implement a state in which, when an option is clicked, the option that has the opposite stat would be selected instead.

For a more specific example, lets say you're on a block that has options A and B. 
Option A modifies "stat1" by +5.
Option B modifies "stat2" by +5.
These are opposite stats.
If this state is active, this custom script will "know" that option A will give you "stat1" +5, so when you select option A, it will act as if you have selected Option B.

I know this is a very specific thing, but I hope you can advise me on it.

**Comment by TheEmbracedOne** [comment]
OK so I figured out the system, but I have a problem at the end where I'm calling the other block; it just gets stuck on the first Say command:
![Unity_FrIJ50vsE1](https://user-images.githubusercontent.com/16197906/91646842-41733880-ea43-11ea-9a54-0dbc50f7ed2e.png)

I'm finding the appropriate block by name and then use Block.StartExecution()

Any ideas?

**Comment by TheEmbracedOne** [comment]
Im happy to report I figured it out.
The initial logic and implementation actually works perfectly, but I had to separate some of the functionality into states (state machine). Now, calling things in the correct order with a little bit of a time delay inbetween, it works perfectly as I needed it!

In short what I do is;
1) Get the connected blocks
2) then when a menu option is clicked, I send an event trigger to the state machine
3) take note of the new executing block, get its description and based on said description, determine an "override string"
4) stop current executing block
5) find the block whose description matches "override string" among connected blocks from step 1
6) execute said found block
